pkg/webui/console/views/application-api-key-add/application-api-key-add.js

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Container, Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+
+import PageTitle from '@ttn-lw/components/page-title'
+import Breadcrumb from '@ttn-lw/components/breadcrumbs/breadcrumb'
+import { withBreadcrumb } from '@ttn-lw/components/breadcrumbs/context'
+
+import { ApiKeyCreateForm } from '@console/components/api-key-form'
+
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+import PropTypes from '@ttn-lw/lib/prop-types'
+
+@withBreadcrumb('apps.single.api-keys.add', function(props) {
+  const appId = props.appId
+  return <Breadcrumb path={`/applications/${appId}/api-keys/add`} content={sharedMessages.add} />
+})
+export default class ApplicationApiKeyAdd extends React.Component {
+  static propTypes = {
+    appId: PropTypes.string.isRequired,
+    createApiKey: PropTypes.func.isRequired,
+    navigateToList: PropTypes.func.isRequired,
+    pseudoRights: PropTypes.rights.isRequired,
+    rights: PropTypes.rights.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+
+    const { createApiKey, appId } = props
+
+    this.createApplicationKey = key => createApiKey(appId, key)
```

Comment:
Please bind `appId` in [mergeProps](https://react-redux.js.org/api/connect#mergeprops-stateprops-dispatchprops-ownprops--object).

```js
    (stateProps, dispatchProps, ownProps) => ({
      ...stateProps,
      ...dispatchProps,
      ...ownProps,
      createApplicationApiKey: key => dispatchProps.createApiKey(stateProps.appId, key),
    }),
```

Reply:
Same applies for `getApplicationsRightsList` and `navigateToList`